modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/resource/v1_0/ObjectEntryResourceImpl.java

```diff
@@ -617,6 +630,34 @@ private void _loadObjectDefinition(Map<String, Serializable> parameters)
 		throw new NotFoundException("Missing parameter \"objectDefinitionId\"");
 	}
 
+	private void _validateObjectProperties(
+		ObjectEntry objectEntry, ObjectDefinition objectDefinition,
+		Locale locale) {
+
+		Stream<String> stream = objectEntry.getProperties(
+		).keySet(
+		).stream();
+
+		List<String> propertyNames = stream.collect(Collectors.toList());
+
+		for (ObjectField objectField :
+				_objectFieldLocalService.getObjectFields(
+					objectDefinition.getObjectDefinitionId())) {
+
+			propertyNames.remove(objectField.getName());
+		}
+
+		if (!propertyNames.isEmpty()) {
+			throw new BadRequestException(
+				String.format(
+					"Object %s does not have the following field(s): %s",
+					objectDefinition.getLabel(locale),
+					ListUtil.toString(
+						propertyNames, (String)null,
+						StringPool.COMMA_AND_SPACE)));
+		}
+	}
```

Comment:
@locpham97 @TaiPhamLR to follow the `Objects` pattern, we should perform the validations in the service layer, which in this case is the `ObjectEntryLocalServiceImpl`.

That said, it looks like we're already validating this scenario [here](https://github.com/liferay/liferay-portal/blob/347d0d9e32ce975c2f9ac0cc6070b73ebb1a50c2/modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java#L3342-L3361), we just need to throw and handle the exception appropriately.

We also need to pass all the `properties` to the service layer, and I believe we can achive this [here](https://github.com/liferay/liferay-portal/blob/347d0d9e32ce975c2f9ac0cc6070b73ebb1a50c2/modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/DefaultObjectEntryManagerImpl.java#L993-L1002), by doing something like:

```
Map<String, Serializable> values = new HashMap<>();

Map<String, Object> properties = objectEntry.getProperties();

for (Map.Entry<String, Object> entry : properties.entrySet()) {
	values.put(entry.getKey(), (Serializable)entry.getValue());
}
```

Could you check if this approach works?

Also, it is recommended to avoid using stream because of this [message](https://liferay.slack.com/archives/CL69D8RUG/p1649806046880829).

Thanks!